src/meta/src/model/system_param.rs

```diff
@@ -0,0 +1,189 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+
+use async_trait::async_trait;
+use risingwave_pb::meta::SystemParams;
+
+use super::{MetadataModel, MetadataModelError, MetadataModelResult};
+use crate::storage::{MetaStore, Snapshot, Transaction};
+
+const SYSTEM_PARAM_CF_NAME: &str = "cf/system_params";
+
+const BARRIER_INTERVAL_MS_KEY: &str = "barrier_interval_ms";
```

Comment:
If we introduce some new params in the future, we need to add them in multi places, like proto definition, SystemParams initialize, here and `system_param_from_kv` function. I am worried that it will be easy to miss but don't have any better solution.

Reply:
+1. What's the benefit of storing each item with a human-readable key? This also breaks the backward compatibility mechanism of the protobuf message itself.

Reply:
See also https://github.com/risingwavelabs/rfcs/pull/34#issuecomment-1415117338. My motivation is to allow other system components or human users to initialize/modify the value in some special cases, for example, fill the `state_store_url` before Meta starts. 

On the other hands, as described in the RFC we hope to allow users to set/get/list values via SQL commands, which makes it inevitable to introduce a human-readable representation. At least this code line is necessary to implement SQL commands.

Reply:
> At least this code line is necessary to implement SQL commands.

I can't promise this is correct. Actually, what we doing now is just a self-made and error-prone version of the protobuf reflection. These codes can be replaced by a general `message.descriptor().all_fields()` without repeating the keys anywhere.

Reply:
We still need to match `all_fields()` against existing ones to know which field to change.

Reply:
> > At least this code line is necessary to implement SQL commands.
> 
> I can't promise this is correct. Actually, what we doing now is just a self-made and error-prone version of the protobuf reflection (note that the `Params` itself is a protobuf message). These codes can be replaced by a general `message.descriptor().all_fields()` without repeating the keys anywhere.

Well, technically the protobuf message definition contains all the necessary to do such things, like implementing set/get/list, but it's not supposed to do so. If `prost` can generate these code for us, that's good - we can easily build a `store_as_human_readable_kvs` function on it. But there don't seem to be such features?

Reply:
If we really want to do that, maybe we can give https://crates.io/crates/prost-reflect a try.

Reply:
As discussed offline, these constants are inevitable. We might use some macro to simplify the code.